src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtils.java

```diff
@@ -0,0 +1,236 @@
+                                                    CaseTypeDefinition caseTypeDefinition,
+                                                    CaseDataService caseDataService) {
+        this.caseDataService = caseDataService;
+        this.caseTypeDefinition = caseTypeDefinition;
+
+        removeCCDAllCasesAccessFromCaseAccessGroups(caseDetails);
+
+        List<AccessTypeRoleDefinition> accessTypeRoleDefinitions =
+            caseTypeDefinition.getAccessTypeRoleDefinitions();
+        List<AccessTypeRoleDefinition> filteredAccessTypeRolesDefinitions = filterAccessRoles(caseDetails,
+            accessTypeRoleDefinitions);
+        if (filteredAccessTypeRolesDefinitions != null) {
+            List<CaseAccessGroupWithId> caseAccessGroupWithIds = new ArrayList<>();
+            for (AccessTypeRoleDefinition acd : filteredAccessTypeRolesDefinitions) {
+                JsonNode caseAssignedRoleFieldNode = findOrganisationPolicyNodeForCaseRole(caseDetails,
+                    acd.getCaseAssignedRoleField());
+                if (caseAssignedRoleFieldNode != null) {
+                    String orgIdentifier = caseAssignedRoleFieldNode.get(ORGANISATION)
+                        .get(ORGANISATIONID).textValue();
+
+                    if (orgIdentifier != null && !orgIdentifier.isEmpty()) {
+                        String caseGroupID = acd.getCaseAccessGroupIdTemplate()
+                            .replace(ORG_IDENTIFIER_TEMPLATE, orgIdentifier);
+
+                        CaseAccessGroup caseAccessGroup = CaseAccessGroup.builder().caseAccessGroupId(caseGroupID)
+                            .caseAccessGroupType(CCD_ALL_CASES).build();
+                        CaseAccessGroupWithId caseAccessGroupWithId = CaseAccessGroupWithId.builder()
+                            .caseAccessGroup(caseAccessGroup).id(UUID.randomUUID().toString()).build();
+
+                        caseAccessGroupWithIds.add(caseAccessGroupWithId);
+                    }
+                }
+            }
+            setUpModifiedCaseAccessGroups(caseDetails, caseAccessGroupWithIds);
+        }
+
+    }
+
+    private void setUpModifiedCaseAccessGroups(CaseDetails caseDetails,
+                                               List<CaseAccessGroupWithId> caseAccessGroupWithIds) {
+        ObjectMapper mapper = new ObjectMapper();
+        if (!caseAccessGroupWithIds.isEmpty()) {
+            JsonNode caseAccessGroupWithIdsNode = mapper.convertValue(caseAccessGroupWithIds, JsonNode.class);
+            if (caseDetails.getData().get(CASE_ACCESS_GROUPS) != null) {
+                JsonNode caseDataCaseAccessGroup = caseDetails.getData().get(CASE_ACCESS_GROUPS);
+
+                String mergedValue = null;
+                if (caseDataCaseAccessGroup != null && !caseDataCaseAccessGroup.isEmpty()) {
+                    mergedValue = caseDataCaseAccessGroup.toString() + caseAccessGroupWithIdsNode.toString();
+                    mergedValue = mergedValue.replace("][",",");
```

Comment:
I assume this is because we're doing a `String` concatenation above of two arrays - it'd probably be better/easier to use the Jackson methods to merge rather than us having to write `JsonNode` to `String`, doing some manual processing then converting back to `JsonNode`.